bingo.js

```diff
@@ -562,10 +562,23 @@ function generateNewUrlParam()
 	return DIFFICULTY + "-" + hidden + "-" + streamerMode + "-" + VERSION.id + "_" + SEED;
 }
 
+function generateNewUrlParamWithProgress(squareProgress)
+{
+	if (squareProgress === undefined) {
+		if (gup(URL_PARAM_PROGRESS)) {
+			squareProgress = generateSquareProgress();
+		} else {
+			return "?s=" + generateNewUrlParam();
+		}
+	}	
+	
+	return "?s=" + generateNewUrlParam() + "&progress=" + JSON.stringify(squareProgress);
```

Comment:
This function might be more readable if this part: `"?s=" + generateNewUrlParam()` – wasn't repeated twice. Maybe something like this:

```js
var urlParams = "?s=" + generateNewUrlParam();
if (squareProgress === undefined) {
	if (gup(URL_PARAM_PROGRESS)) {
		squareProgress = generateSquareProgress();
	}
}
if (squareProgress) {
	urlParams += "&progress=" + JSON.stringify(squareProgress);
}
return urlParams;
```


?